src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -1,30 +1,51 @@
 import { Popup } from '../Popup/Popup'
 import './PopupWithForm.css'
+import { useForm } from '../../formHooks/useForm'
+import { useEffect, useRef } from 'react'
+
+export function PopupWithForm({ isFormLoading, globalErrorMessage, onSubmit, btnText, redirectText, onClick, isSignUpOpen, onClose, isOpen, isSignInOpen, preSpanText, setEmail, setPassword, setName }) {
```

Comment:
`PopupWithForm` should be a unversal component for any popups with a form. So, you need to make it the way it was in your 14th sprint. The `children` should contain the whole neccessary markup.